File: hw/armijos/u3/hw30StrategyandObserver/StrategyPattern/src/main/java/ec/edu/espe/sortingapp/controller/BubbleSort.java
package ec.edu.espe.sortingapp.controller;

/**
 *
 * @author Enzo Armijos, LogicLegion, DCCO-ESPE
 */
public class BubbleSort implements SortingStrategy {

    @Override
    public int[] sort(int[] data) {
        int n = data.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (data[j] > data[j + 1]) {
                    swap(data, i, j);
                }
            }
        }
        return data;
    }

    private void swap(int[] arr, int i, int j) {
        int temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
    }
    
}

File: exams/segarra/u3/PracticalExam/ec.edu.espe.stateypattern.model/BubbleSort.js
class BubbleSort {
    sort(data) {
        let n = data.length;
        for (let i = 0; i < n - 1; i++) {
            for (let j = 0; j < n - i - 1; j++) {
                if (data[j] > data[j + 1]) {
                    let temp = data[j];
                    data[j] = data[j + 1];
                    data[j + 1] = temp;
                }
            }
        }
        return data;
    }
}